### PR TITLE
Use `dart` CLI and move `dart_test.browser_aggregate.yaml` config to package root

### PR DESCRIPTION
**Pull request by evanweible-wf**
Now that we can use `$package$` as a synthetic input, we can write outputs to the package root instead of one of the dirs like `test/`. Since we're going to release a major version of this package for NNBD anyway, it's an opportunity to move the `dart_test.browser_aggregate.yaml` output to the package root since it makes more sense there alongside the main `dart_test.yaml` file.

This PR also updates everything to use the new `dart` CLI.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by evanweible-wf** [comment]
CI failures on Dart stable and above are expected due to language version mismatch between pkg:analyzer and the Dart SDK. This will be fixed when we migrate to NNBD.

File: build.yaml [hunk 1]
```diff
@@ -10,6 +10,9 @@ builders:
     builder_factories:
       - testHtmlBuilder
     build_extensions:
+      $test$:
+        - templates/default_template.html
+        - test_html_builder_config.json
       $package$:
         - test/templates/default_template.html
         - test/test_html_builder_config.json
```
Comment (todbachman-wf) [inline build.yaml:18]: What is the effect of listing these files under both `$test$` and `$package$`? I would have expected we'd only need them listed under one or the other not both.

**Review by todbachman-wf — APPROVED** [review]

File: build.yaml (hunk 1 above)
Comment (evanweible-wf) [reply build.yaml:18]: Oh good catch, I think this was a bad merge conflict resolution!

Comment (evanweible-wf) [reply build.yaml]: Should be fixed now

**Review by todbachman-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
Merging into v3 branch.

**Pull request merged**